src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/PatientMapper.java

```diff
@@ -62,13 +64,9 @@ public Patient mapToPatient(final Message message) {
 
     private void mapAddress(final UnstructuredAddress unstructuredAddress, final Patient patient) {
         final Address address = new Address();
-
-        final String[] addressLines = unstructuredAddress.getAddressLines();
-        if (addressLines != null && addressLines.length > 0) {
-            Arrays.stream(addressLines).forEach(address::addLine);
-        }
-        address.setPostalCode(unstructuredAddress.getPostCode());
-
+        final Optional<String[]> addressLines = unstructuredAddress.getAddressLines();
+        addressLines.map(Arrays::stream).ifPresent(stream -> stream.forEach(address::addLine));
```

Comment:
If you do action the previous suggestion to return the empty array, this becomes simply
```suggestion
        Arrays.stream(unstructuredAddress.getAddressLines()).forEach(address::addLine);
```

Reply:
Or in the previous suggestion return a List instead of an array (which in case of no values will be `Collections.emptyList()`)